### PR TITLE
Fix delete command for paused kic driver with containerd/crio runtime

### PR DESCRIPTION
**Pull request by ilya-zuyev**
This PR fixes an issue when delete operation hangs forever for paused minikube when (container-runtime=containerd or container-runtime=crio) and driver=docker
Fixes https://github.com/kubernetes/minikube/issues/11332

Before:
```
➜  minikube git:(master) m start --container-runtime=containerd
😄  minikube v1.20.0 on Darwin 11.3.1
✨  Automatically selected the docker driver. Other choices: virtualbox, ssh
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
🔥  Creating docker container (CPUs=2, Memory=4000MB) ...
📦  Preparing Kubernetes v1.20.2 on containerd 1.4.4 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔗  Configuring CNI (Container Networking Interface) ...
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default



➜  minikube git:(master) m pause
⏸️  Pausing node minikube ... 
⏯️  Paused 1 containers in: kube-system, kubernetes-dashboard, storage-gluster, istio-operator


➜  minikube git:(master) m delete 
🔥  Deleting "minikube" in docker ...
🔥  Deleting container "minikube" ...
[hangs]
^C
➜  minikube git:(master) 
```

After:
```
➜  minikube git:(ilyaz/fix_delete_paused) ✗ m start --container-runtime=containerd
😄  minikube v1.20.0 on Darwin 11.3.1
✨  Automatically selected the docker driver. Other choices: virtualbox, ssh
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
🔥  Creating docker container (CPUs=2, Memory=4000MB) ...
📦  Preparing Kubernetes v1.20.2 on containerd 1.4.4 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔗  Configuring CNI (Container Networking Interface) ...
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default


➜  minikube git:(ilyaz/fix_delete_paused) ✗ m pause
⏸️  Pausing node minikube ... 
⏯️  Paused 1 containers in: kube-system, kubernetes-dashboard, storage-gluster, istio-operator


➜  minikube git:(ilyaz/fix_delete_paused) ✗ m delete
🔥  Deleting "minikube" in docker ...
🔥  Deleting container "minikube" ...
🔥  Removing /Users/izuyev/.minikube/machines/minikube ...
💀  Removed all traces of the "minikube" cluster.
```

### CODE REVIEW

**Comment by ilya-zuyev** [comment]
http://storage.googleapis.com/minikube-builds/logs/11504/773500b/KVM_Linux_containerd.html#pass_TestPause%2fserial%2fDeletePaused 
Test fixed

File: cmd/minikube/cmd/delete.go [hunk 1]
```diff
@@ -295,6 +303,48 @@ func deleteProfile(ctx context.Context, profile *config.Profile) error {
 	return nil
 }
 
+func unpauseIfNeeded(profile *config.Profile) error {
+	// there is a known issue with removing kicbase container with paused containerd containers inside
+	// unpause it before we delete it
+	if profile.Config.KubernetesConfig.ContainerRuntime != "containerd" {
```
Comment (medyagh) [inline cmd/minikube/cmd/delete.go:309]: crio also has this problem., we can check if Not Docker

**Review by medyagh — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/delete.go [hunk 2]
```diff
@@ -210,23 +210,28 @@ func DeleteProfiles(profiles []*config.Profile) []error {
...
+}
 
-			if !profile.IsValid() || (loadErr != nil || !mm.IsValid()) {
-				invalidProfileDeletionErrs := deleteInvalidProfile(profile)
-				if len(invalidProfileDeletionErrs) > 0 {
-					errs = append(errs, invalidProfileDeletionErrs...)
-				}
-			} else {
-				errs = append(errs, err)
+func deleteProfileTimeout(profile *config.Profile) []error {
```
Comment (medyagh) [inline cmd/minikube/cmd/delete.go:218]: thanks for breaking down this long code to more modular one, this is more readable !

Comment (ilya-zuyev) [reply cmd/minikube/cmd/delete.go:218]: there was a small issue with `defer cancel()` actually. When called inside the `for` loop, cancel() is deferred till the end of the function, not the loop iteration

File: cmd/minikube/cmd/delete.go (hunk 1 above)
Comment (ilya-zuyev) [reply cmd/minikube/cmd/delete.go:309]: fixed

**Pull request merged**